### ISSUE TITLE
Minibatch size, number of iterations, epoch 

### ISSUE THREAD

**Issue by bamba518** [kaldi-asr/kaldi#2526]
I have a 1080 card with 12 mb of ram, and get memory issues with most of the dnn set ups
 --trainer.num-chunk-per-minibatch=128, 64 seem to be too big?
If I change this to a lower number like 16, 8 then would i need to adjust  epoch number and number of iterations accordingly to train to the same amount?

**Comment by danpovey** [comment]
Probably because you are trying to share the GPU between multiple jobs.
You should set exclusive mode (nvidia-smi -c 3) and use the GPU for one job
at a time.  You can change --use-gpu=wait to --use-gpu=yes to
chain/train.py or whatever training script it is.

**Comment by danpovey** [comment]
and see kaldi-asr.org/forums.html for how to ask questions.

**Comment by 13265170340** [comment]
How to set up end-to-end minibatch
2019-08-08 10:20:30,682 [steps/nnet3/chain/e2e/train_e2e.py:460 - train - INFO ] Iter: 0/8711   Jobs: 2   Epoch: 0.00/4.0 (0.0% complete)   lr: 0.002000   
run.pl: job failed, log is in /data/wdz_egs/exp/chain/e2e_tdnnf_att_/log/train.0.2.log
2019-08-08 10:24:19,036 [steps/libs/common.py:236 - background_command_waiter - ERROR ] Command exited with status 1: run.pl --gpu 1 /data/wdz_egs/exp/chain/e2e_tdnnf_att_/log/train.0.2.log                     nnet3-chain-train --use-gpu=wait                      --apply-deriv-weights=False                     --l2-regularize=5e-05 --leaky-hmm-coefficient=0.1                       --xent-regularize=0.1                                          --print-interval=10 --momentum=0.0                     --max-param-change=1.414213562373095                     --backstitch-training-scale=0.0                     --backstitch-training-interval=1                     --l2-regularize-factor=0.5                      --srand=0                     "nnet3-am-copy --raw=true --learning-rate=0.002 --scale=1.0 /data/wdz_egs/exp/chain/e2e_tdnnf_att_/0.mdl - |nnet3-copy --edits='set-dropout-proportion name=* proportion=0.0' - - |" /data/wdz_egs/exp/chain/e2e_tdnnf_att_/den.fst                     "ark,bg:nnet3-chain-copy-egs                          --frame-shift=2                         ark:/data/wdz_egs/exp/chain/e2e_tdnnf_att_/egs/cegs.2.ark ark:- |                         nnet3-chain-shuffle-egs --buffer-size=5000                         --srand=0 ark:- ark:- | nnet3-chain-merge-egs                         --minibatch-size=8,4,2,1,1 ark:- ark:- |"                     /data/wdz_egs/exp/chain/e2e_tdnnf_att_/1.2.raw
run.pl: job failed, log is in /data/wdz_egs/exp/chain/e2e_tdnnf_att_/log/train.0.1.log
2019-08-08 10:26:25,217 [steps/libs/common.py:236 - background_command_waiter - ERROR ] Command exited with status 1: run.pl --gpu 1 /data/wdz_egs/exp/chain/e2e_tdnnf_att_/log/train.0.1.log                     nnet3-chain-train --use-gpu=wait                      --apply-deriv-weights=False                     --l2-regularize=5e-05 --leaky-hmm-coefficient=0.1                      --write-cache=/data/wdz_egs/exp/chain/e2e_tdnnf_att_/cache.1  --xent-regularize=0.1                                          --print-interval=10 --momentum=0.0                     --max-param-change=1.414213562373095                     --backstitch-training-scale=0.0                     --backstitch-training-interval=1                     --l2-regularize-factor=0.5                      --srand=0                     "nnet3-am-copy --raw=true --learning-rate=0.002 --scale=1.0 /data/wdz_egs/exp/chain/e2e_tdnnf_att_/0.mdl - |nnet3-copy --edits='set-dropout-proportion name=* proportion=0.0' - - |" /data/wdz_egs/exp/chain/e2e_tdnnf_att_/den.fst                     "ark,bg:nnet3-chain-copy-egs                          --frame-shift=1                         ark:/data/wdz_egs/exp/chain/e2e_tdnnf_att_/egs/cegs.1.ark ark:- |                         nnet3-chain-shuffle-egs --buffer-size=5000                         --srand=0 ark:- ark:- | nnet3-chain-merge-egs                         --minibatch-size=8,4,2,1,1 ark:- ark:- |"                     /data/wdz_egs/exp/chain/e2e_tdnnf_att_/1.1.raw
